Assets/MRTK/Providers/LeapMotion/LeapMotionDeviceManager.cs

```diff
@@ -0,0 +1,240 @@
+
+        /// <summary>
+        /// List of hands that are currently in frame and detected by the leap motion controller. If there are no hands in the current frame, this list will be empty.
+        /// </summary>
+        public List<Hand> CurrentHandsDetectedByLeap => LeapMotionServiceProvider.CurrentFrame.Hands;
+
+        // This value can only be set in the profile, the default is LeapControllerOrientation.Headset.
+        private LeapControllerOrientation leapControllerOrientation => SettingsProfile.LeapControllerOrientation;
+
+        /// <summary>
+        /// Adds an offset to the game object with LeapServiceProvider attached.  This offset is only applied if the leapControllerOrientation
+        /// is LeapControllerOrientation.Desk and is necessary for the hand to appear in front of the main camera. If the leap controller is on the 
+        /// desk, the LeapServiceProvider is added to the scene instead of the LeapXRServiceProvider. The anchor point for the position of the leap hands is 
+        /// the position of the game object with the LeapServiceProvider attached.
+        /// </summary>
+        private Vector3 leapHandsOffset => SettingsProfile.LeapControllerOffset;
+
+        /// <summary>
+        /// Dictionary to capture all active leap motion hands detected.
+        /// </summary>
+        private readonly Dictionary<Handedness, LeapMotionArticulatedHand> trackedHands = new Dictionary<Handedness, LeapMotionArticulatedHand>();
+
+        private AttachmentHand leftAttachmentHand = null;
+        private AttachmentHand rightAttachmentHand = null;
+
+        private static readonly ProfilerMarker UpdatePerfMarker = new ProfilerMarker("[MRTK] LeapMotionDeviceManager.Update");
+
+        public override void Enable()
+        {
+            base.Enable();
+
+            if (leapControllerOrientation == LeapControllerOrientation.Headset)
+            {
+                // If the leap controller is mounted on a headset then add the LeapXRServiceProvider to the scene
+                // The LeapXRServiceProvider can only be attached to a camera 
+                LeapMotionServiceProvider = CameraCache.Main.gameObject.AddComponent<LeapXRServiceProvider>();
+            }
+
+            if (leapControllerOrientation == LeapControllerOrientation.Desk)
+            {
+                // Create a separate gameobject if the leap controller is on the desk
+                GameObject leapProvider = new GameObject("LeapProvider");
+
+                // The LeapServiceProvider does not need to be attached to a camera, but the location of this gameobject is the anchor for the desk hands
+                LeapMotionServiceProvider = leapProvider.AddComponent<LeapServiceProvider>();
+
+                // Follow the transform of the main camera by adding the service provider as a child of the main camera
+                leapProvider.transform.parent = CameraCache.Main.transform;
```

Comment:
Does this guarantee the provider will be at a local position of (0,0,0) under the camera to start with? Or will it be slightly offset if the camera has moved?

Reply:
This guarantees that the hands will appear, in front and in view of the camera regardless of the camera starting position.  The offset is calculated relative to the camera position, so if the camera has moved then the hands will move with it.